addons/product/models/product_attribute.py

```diff
@@ -148,9 +148,13 @@ def unlink(self):
         for pav in self:
             if pav.is_used_on_products:
                 raise UserError(
-                    _("You cannot delete the value %s because it is used on the following products:\n%s") %
+                    _("You cannot delete the value %s because it is used on the following products:\n%s\nIf the value has been associated to other product in the past, you will not be able to deleted it.") %
                     (pav.display_name, ", ".join(pav.pav_attribute_line_ids.product_tmpl_id.mapped('display_name')))
                 )
+            linked_products = pav.env['product.template.attribute.value'].search([('product_attribute_value_id', '=', pav.id)]).with_context(active_test=False).ptav_product_variant_ids
+            unlinkable_products = linked_products._filter_to_unlink()
+            if linked_products != unlinkable_products:
+                raise UserError(_("Cannot delete value %s because it was used in some products", pav.display_name))
```

Comment:
This is a new translated string, you have to update the .pot files when you modify translated strings in stable version (to request new translations on Transifex for the new/modified string, and create corresponding records in database).

Reply:
:ok_hand: